modules/navier_stokes/src/physics/WCNSLinearFVTurbulencePhysics.C

```diff
@@ -0,0 +1,324 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WCNSLinearFVTurbulencePhysics.h"
+#include "WCNSFVFlowPhysics.h"
+#include "WCNSFVFluidHeatTransferPhysics.h"
+#include "WCNSFVScalarTransportPhysics.h"
+#include "WCNSFVCoupledAdvectionPhysicsHelper.h"
+#include "INSFVTurbulentViscosityWallFunction.h"
+#include "INSFVTKESourceSink.h"
+#include "NSFVBase.h"
+
+registerWCNSFVTurbulenceBaseTasks("NavierStokesApp", WCNSLinearFVTurbulencePhysics);
+registerMooseAction("NavierStokesApp", WCNSLinearFVTurbulencePhysics, "add_functor_material");
+
+InputParameters
+WCNSLinearFVTurbulencePhysics::validParams()
+{
+  InputParameters params = WCNSFVTurbulencePhysicsBase::validParams();
+  params.addClassDescription(
+      "Define a turbulence model for an incompressible or weakly-compressible Navier Stokes "
+      "flow with a linear finite volume discretization");
+
+  params.addParam<std::vector<SolverSystemName>>(
+      "system_names",
+      {"TKE_system", "TKED_system"},
+      "Names of the linear solver systems for each equation. Default is set for K-Epsilon and "
+      "should be adapted for other models");
+
+  // Not an option
+  params.addParam<bool>("mu_t_as_aux_variable",
+                        true,
+                        "Whether to use an auxiliary variable instead of a functor material "
+                        "property for the turbulent viscosity");
+  params.suppressParameter<bool>("mu_t_as_aux_variable");
+  params.suppressParameter<bool>("turbulent_viscosity_two_term_bc_expansion");
+
+  // Could be implemented when boundary conditions are implemented in turbulence physics
+  params.suppressParameter<bool>("tke_two_term_bc_expansion");
+  params.suppressParameter<bool>("tked_two_term_bc_expansion");
+
+  // Not implemented
+  params.suppressParameter<MooseEnum>("wall_treatment_T");
+  params.suppressParameter<MooseEnum>("tke_face_interpolation");
+  params.suppressParameter<MooseEnum>("tked_face_interpolation");
+
+  // LinearFV-specific parameters
+  params.addParam<bool>(
+      "use_nonorthogonal_correction",
+      true,
+      "Whether to use a non-orthogonal correction. This can potentially slow down convergence "
+      ", but reduces numerical dispersion on non-orthogonal meshes. Can be safely turned off on "
+      "orthogonal meshes.");
+  params.addParamNamesToGroup("use_nonorthogonal_correction", "Numerical scheme");
+  return params;
+}
+
+WCNSLinearFVTurbulencePhysics::WCNSLinearFVTurbulencePhysics(const InputParameters & parameters)
+  : WCNSFVTurbulencePhysicsBase(parameters)
```

Comment:
At some point it would be nice to create different physics for the different flavors of turbulence. MixingLenghtTurbulencePhysics, KEpsilonturbulence physics etc. Maybe not for now just sometime in the future.

Reply:
I was thinking we might get away with a single class. It works fine for WCNSFV. I think having a second model might be the test of that